File: 1301-1400/1366-Rank Teams by Votes/1366-Rank Teams by Votes.py
class Solution:
    def rankTeams(self, votes: List[str]) -> str:
        rank = {c: [0] * len(votes[0]) + [c] for c in votes[0]}
        for vote in votes:
            for i, c in enumerate(vote):
                rank[c][i] -= 1
                
        return ''.join(sorted(votes[0], key=rank.get))

File: 1301-1400/1366-Rank Teams by Votes/1366-Rank Teams by Votes.cpp
class Solution {
public:
    string rankTeams(vector<string>& votes) {
        vector<vector<int>> rank(26, vector<int>(27));
        for (char c : votes[0]) {
            rank[c - 'A'][26] = c;
        }
        for (string& vote : votes) {
            for (int i = 0; i < vote.size(); ++i) {
                --rank[vote[i] - 'A'][i];
            }
        }
        sort(rank.begin(), rank.end());
        string result;
        for (int i = 0; i < votes[0].size(); ++i) {
            result += rank[i][26];
        }
        return result;
    }
};
